### ISSUE TITLE
Incorrect detection of properties of t.context as AVA functions

### ISSUE THREAD

**Issue by GMartigny** [avajs/eslint-plugin-ava#223]
[<img alt="Issuehunt badges" src="https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/223/badge.svg" />](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/223)

Hi,
When a property of `t.context` match an existing AVA function (say `end`), the eslint plugin fire an error for incorrect use of this function.
Consider:
```js
test.beforeEach((t) => {
    t.context = new Vector(1, 2);
});
test("constructor", (t) => {
    t.is(t.context.start, 1);
    t.is(t.context.end, 2);		// <= `t.end()` should only be used inside of `test.cb()`  ava/no-invalid-end
});
```

I have search the source and it appear that only `end` and `skip` raised the issue.

> There is a $40.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/223).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

**Comment by novemberborn** [comment]
I suppose `nameOfRootObject` doesn't do what we think it does: https://github.com/avajs/eslint-plugin-ava/blob/c5a77bb6c9ce89471acad2aa603ab964f71506cb/rules/no-invalid-end.js#L17

It's used in some other rules too: https://github.com/avajs/eslint-plugin-ava/search?q=nameofrootobject&unscoped_q=nameofrootobject

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $40.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/49722492/issues/223) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/49722492/issues/223).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.

**Comment by IssueHuntBot** [comment]
@sindresorhus has rewarded $36.00 to @gmartigny. [See it on IssueHunt](https://issuehunt.io/repos/49722492/issues/223)

- :moneybag: Total deposit: $40.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $4.00
